apps/explorer/src/components/Layout/index.tsx

```diff
@@ -41,7 +41,12 @@ export function Layout() {
 				/*autoConnect={false}*/
 				enableUnsafeBurner={import.meta.env.DEV}
 			>
-				<RpcClientContext.Provider value={jsonRpcProvider}>
+				<SuiClientProvider
+					networks={NetworkConfigs}
+					createClient={createSuiClient}
+					network={network as Network}
+					onNetworkChange={setNetwork}
```

Comment:
I'm curious if we're accounting for this specific edge case where the selected network might change from browser navigation as opposed to calling `selectNetwork` when the network changes via the dropdown selector. I know handling the whole network selection piece is tricky from looking at your previous comments and past experience 😛

1. I go to `suiexplorer.com/?network=testnet`, so we pass `network` through to the `SuiClientProvider` component. We have `network=testnet`, `selectedNetwork=testnet`, and `currentNetwork='testnet`.

2. Instead of changing networks via the dropdown selector, I navigate backwards to `suiexplorer.com/?network=mainnet`. Now we have `network=mainnet`, `selectedNetwork=testnet`, and `currentNetwork=mainnet`. 

3. I use this context to access `selectedNetwork` somewhere else in my code, but now it's pointing to the wrong value since it's out of sync with the network I'm on (i.e., `onNetworkChange` was never called)

Reply:
great catch... I totally forgot to update the context after refactoring this.  the selectedNetwork in the context should be `currentNetwork`

Reply:
updated the context to have `network: currentNetwork` instead of `selectedNetwork`